app/models/post.rb

```diff
@@ -18,7 +20,7 @@ def self.create(post_params)
     post = new.tap do |p|
       salty_password = post_params[:has_salt] ? post_params[:salty_password] : ''
       p.body = self.ciphered(post_params[:body], salty_password)
-      p.expired_at = time_to_sec(post_params[:expired_at].to_i) if post_params[:expired_at]
+      p.expired_at = time_to_sec(EXPIRE_IN[post_params[:expired_at].to_i]) if post_params[:expired_at]
```

Comment:
as described in the chat this is incompatible with the frontend. `EXPIRE_IN` is an array so when frontend passes `12` or `24` , the value captured here is `  EXPIRE_IN[12]` . This means the code tries to look for a value at index 12 in `EXPIRE_IN` array. 

The code here is not validating, it's simply selecting a value from the array and expects an index not a value. The array itself is not immutable. 

The right way to validate this would be -

```
EXPIRE_IN = [2, 12, 24].freeze
....
return if !EXPIRE_IN.includes?([post_params[:expired_at].to_i])
.....
```